### PR TITLE
Adding Category to Test Category mapping for ListFullyQualifiedTests

### PR DESCRIPTION
**Pull request by nigurr**
Currently ListFullyQualifiedTests doesn't filter tests based on TestCategory for Nunit/Xunit adapters

### CODE REVIEW

**Comment by smadala** [comment]
@dotnet-bot please test Windows_NT / Release Build

**Comment by nigurr** [comment]
@dotnet-bot please test Windows_NT / Release Build

File: src/vstest.console/Processors/ListFullyQualifiedTestsArgumentProcessor.cs [hunk 1]
```diff
@@ -436,6 +437,13 @@ private static List<TestCase> GetFilteredTestCases(IEnumerable<TestCase> testCas
                     }
                 }
 
+                //This is hack for NUnit,Xunit to understand test category -> This method is called only for NUnit/Xunit
+                if (!traitDictionary.ContainsKey(TestCategory) && traitDictionary.ContainsKey(Category))
```
Comment (acesiddhu) [inline src/vstest.console/Processors/ListFullyQualifiedTestsArgumentProcessor.cs]: did we test TIA with this change

**Review by acesiddhu — APPROVED** [review]

**Review by prawalagarwal — APPROVED** [review]

**Comment by JunielKatarn** [comment]
Can this switch `ListFullyQualifiedTests` be used without an output file?
I just want to print those to the command line.

**Comment by prawalagarwal** [comment]
@JunielKatarn No this cannot be used without an output file. This flag was introduced for a very specific purpose and we needed it to output to a file back then.

**Comment by JunielKatarn** [comment]
@pranavkm How can I run tests in a binary, showing their fully qualified name?
Some namespaces/test classes may end up with colliding names, which makes reading the test output confusing.

**Pull request merged**